CMakeLists.txt

```diff
@@ -51,19 +51,34 @@ add_custom_target(Download)
 add_custom_target(fletch-build-install)
 
 # Options to control GPU support
-option(fletch_BUILD_WITH_CUDA "Build with CUDA support" FALSE)
+option(fletch_ENABLE_CUDA "Build with CUDA support" FALSE)
 
-if (fletch_BUILD_WITH_CUDA)
+if (fletch_ENABLE_CUDA)
   find_package( CUDA QUIET REQUIRED )
 
-  option(fletch_BUILD_WITH_CUDNN "Build with CUDNN support" FALSE)
-
-  if (fletch_BUILD_WITH_CUDNN)
-    set( CUDNN_ROOT "" CACHE PATH "CUDNN root folder" )
-    mark_as_advanced( CUDNN_ROOT )
+  option(fletch_ENABLE_CUDNN "Build with CUDNN support" FALSE)
+  if (fletch_ENABLE_CUDNN)
+    set( CUDNN_ROOT_DIR "" CACHE PATH "CUDNN root folder" )
+    mark_as_advanced( CUDNN_ROOT_DIR )
+    
+    find_library( CUDNN_LIBRARIES
+      NAMES cudnn libcudnn
+      HINTS ${CUDA_TOOLKIT_ROOT_DIR}
```

Comment:
If these are suggested search paths, the standard is to use the PATHS directive, not HINTS. HINTS is for something else....
from: https://cmake.org/cmake/help/v3.0/command/find_library.html

3. Search the paths specified by the HINTS option. These should be paths computed by system introspection, such as a hint provided by the location of another item already found. Hard-coded guesses should be specified with the PATHS option.